doc/src/Glossary.dox

```diff
@@ -592,6 +592,28 @@
  *
  *  @see @ref openassetio.specifications "specifications"
  *
+ * @section trait Trait
+ *
+ * Traits define a concrete aspect of the nature of something. They
+ * area combined to form a @needsref specification - which can then be used
+ * to categorize @ref entity "entities", their relationships or a
+ * host @ref locale.
+ *
+ * Traits also define what data can be queried from an entity using
+ * @needsref entityData.
+ *
+ * Examples of entity traits include `blob`, `cache`, `sequence`,
+ * `geometry`. Relationship traits include `parent`,
+ * `workflow`, `dependency`. Locale traits include `action`, `menu`,
+ * `nodegraph`, `node`.
+ *
+ * @section trait_collection Trait Collections
+ * @see entities_traits_and_specifications
+ *
+ * A trait collection is union one or more traits. They are the basis
```

Comment:
Should trait collections be called Specifications? If we think about Specifications as just being aliases for groups of traits it could be convenient for users to be able to not have to remember a long series of traits.

Reply:
I was wondering if there are two things:

 - A collection of trait names, such as would be passed to `managementPolicy` and `resolve`, ie: `['blob', 'entity', 'image;]`
 - A view on the dict-of-dicts that holds the data for several traits at runtime.

#246 gives an example of how having this separation might be useful.

Reply:
The definition is shared either way, I think the Q is really about whether it makes sense to include the _data_ part of it, for those API calls, which is the subject of #246. If we keep it in, then it creates a lot of ambiguity about what is required behaviour for someone in the system. Sounds like we should try to understand/resolve that problem to answer this Q?

Reply:
Yes, let's come back to this.

Reply:
I've updated #246 to attempt to explain this challenge - what do you think?

Reply:
Right up until now I've assumed that all methods that take a specification may be interested in its properties, it's a revelation to me that there are a bunch of methods where specification properties are irrelevant.  I think making the distinction clear would be really valuable.